### PR TITLE
Dependabot but for Bamboo

### PR DESCRIPTION
**Pull request by hnrkndrssn**
```mermaid
graph TB
    A(Check for new Bamboo Version)
    A -->|No new version| B((Skip))
    A -->|New version| C(Create Issue)
    C -->|Failed| D((Skip))
    C -->|Created| E(Update and Commit POM.xml)
    E -->|No changes| F((Skip))
    E -->|Changes pushed| G(Create Pull Request)
```

### CODE REVIEW

File: .github/workflows/bamboo-compatibility-check.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          currentCompatibleVersion=$(curl -sS -X GET https://marketplace.atlassian.com/rest/2/addons/com.octopus.bamboo/versions/latest | jq '.compatibilities[0].hosting.server.max.version')
+          echo $currentCompatibleVersion
+          # get latest available version of Bamboo
+          latestBambooVersion=$(curl -sS -X GET https://marketplace.atlassian.com/rest/2/products/key/bamboo/versions/latest | jq '.name')
+          echo $latestBambooVersion
+          echo "::set-output name=update-available::true"
+          echo "::set-output name=current-compatible-version::$currentCompatibleVersion"
+          echo "::set-output name=latest-bamboo-version::$latestBambooVersion"
+
+  create-issue:
```
Comment (borland) [inline .github/workflows/bamboo-compatibility-check.yml:41]: typically jobs imply running things in parallel, or sequentially but on different runners (e.g. stage 1 linux stage 2 windows)

This looks like a linear flow here with dependencies between each one; would it be simpler just to have a single job with multiple steps? The conditional/if stuff would work the same

**Review by borland — APPROVED** [review]

File: .github/workflows/bamboo-compatibility-check.yml (hunk 1 above)
Comment (hnrkndrssn) [reply .github/workflows/bamboo-compatibility-check.yml:41]: I had it that way initially but if anything fails, you have to drill into the job then find the step that failed and drill into that. Less clicks having them as separate jobs :grin:

Comment (borland) [reply .github/workflows/bamboo-compatibility-check.yml:41]: LOL fair enough

Comment (hnrkndrssn) [reply .github/workflows/bamboo-compatibility-check.yml:41]: It also makes for a nicer job summary 😄 

image

**Pull request merged**